sparrow_datums/keypoints/keypoints.py

```diff
@@ -122,10 +136,43 @@ def to_heatmap(
             zz_range += 1e-8
         return (zz - zz_min) / zz_range
 
-    def to_keypoint(self, heatmap):
-        ncols = heatmap.shape[-1]
-        print(heatmap.shape, ncols)
-        flattened_keypoint_indices = heatmap.flatten().argmax(-1)
-        x = flattened_keypoint_indices % ncols
-        y = np.floor(flattened_keypoint_indices / ncols)
-        return np.array([x, y], dtype=float)
+    def to_heatmap_array(self, covariance=20):
```

Comment:
@jbencook  I couldn't revive the exact code because I hadn't saved the finished version of the single class design. However, this is very close to what it looked like.
Assuming kp is an instance of the Keypoints class, you can do `heatmap_array = kp.to_heatmap_array()` to get the heatmap in the form of a numpy array.
Then, to get the keypoint back, you have to do;
```
keypoint_back_array = Keypoints.from_heatmap(heatmap_array)# a numpy array
keypoint_back = Keypoints(keypoint_back_array, ptype = kp.ptype, image_h=100, img_w=100) #or use the from dict method.
```
Is this what you have been thinking?

Reply:
Close, I would prefer this interface:

```
keypoint_back = Keypoints.from_heatmap(heatmap_array, ptype=kp.ptype, **kp.metadata_kwargs)
```

Reply:
oh yeah I think it could work that way too.

Reply:
hey @jbencook ,
Just out of curiosity, since, we don't have an init dunder for Numpy subclassing, where exactly does our code handle the `**kp.metadata_kwargs` in the above interface? Does Numpy automatically handle that for us?
![image](https://user-images.githubusercontent.com/63982009/208347476-4ac71276-ce87-4d54-b065-3169ca1ef4b6.png)

Reply:
The full signature is in the `__new__()` method [here](https://github.com/sparrowml/sparrow-datums/blob/27bb011cd1fd20165ecfd55fab91d728d68ad4e1/sparrow_datums/chunk.py#L42-L51). `__new__()` is basically a stand-in for the init constructor. That's an oversimplification bc I can't remember exactly why numpy subclasses use `__new__` instead of `__init__`, but that's how I think about it.